src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -81,14 +81,6 @@ private static int GetNestedProxiesCount(HttpContext context, IFunctionExecution
         {
             context.Items.TryGetValue(ScriptConstants.AzureFunctionsNestedProxyCount, out object nestedProxiesCount);
 
-            if (functionExecution != null && !functionExecution.Descriptor.Metadata.IsProxy && nestedProxiesCount == null)
-            {
-                // HttpBufferingService is disabled for non-proxy functions.
-                var bufferingFeature = context.Features.Get<IHttpBufferingFeature>();
-                bufferingFeature?.DisableRequestBuffering();
-                bufferingFeature?.DisableResponseBuffering();
-            }
```

Comment:
As we discussed, we can't have these changes alone as they would potentially have a significant performance impact because of the buffering middleware introduced by proxies.